### DISCUSSION TITLE
Local operations install and develop

### DISCUSSION THREAD

**Discussion by JiangYuan2899** [directus/directus#10192] [Archive]
There was a issue downloading the latest code from the git repository running locally.

1. git clone https://github.com/directus/directus.git
2. npm install
3. npm run build
4. cp api/example.env api/.env
5. npm run cli bootstrap
6. npm run dev -w @directus/app

In the chrome enter http://localhost:8080
There was no error in the installing

### Errors Shown

20:10:26 [vite] http proxy error:
Error: connect ECONNREFUSED 127.0.0.1:8055
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)

<details>
<summary>Additional Info</summary>

### What version of Directus are you using?

9.1.2

### What version of Node.js are you using?

17.2.0

### What database are you using?

mysql 8.0.18

### What browser are you using?

chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

running locally

</details>

**Comment by azrikahar** [comment]
Moved to discussions since this is more of a setup question, rather than a inherent bug in Directus 👍

Before you run step 6, which is `npm run dev -w @directus/app`, make sure to run `npm run dev -w directus` as well, since the app is just the frontend client which still needs to communicate with the api to actually "work" (depends on context here, hence the quotes). 

Technically this is where it's failing (since Vite _tries_ to proxy localhost:8055 but it couldn't actually achieve it):

https://github.com/directus/directus/blob/481d03df7d4aaa78e0eb78b05aed63d5f49bfbbe/app/vite.config.js#L127-L132

You can also refer what I suggested by checking the docs in case you weren't aware of it: https://docs.directus.io/contributing/running-locally/#_7-start-the-development-server, but if it's not clear, do let us know how we can improve on it. Hope that helps!

**Reply by rijkvanzanten** [reply]
Also make sure to run `npm run build` once to build all the other sub packages 👍🏻

**Comment by Pachat** [comment]
I followed the path in the OP:

1. git clone https://github.com/directus/directus.git
2. npm install
3. npm run build
4. cp api/example.env api/.env  *
6. npm run cli bootstrap
7. npm run dev -w @directus/app

_* Commented PG Database lines and decommented SQLite_

There was no error in the installing
It gave admin email and password

Then on Firefox, localhost:8080/ sent me to [localhost:8080/admin/](url) which after minutes, shows a blank page.
or on refresh, stay waiting endlessly to find the page /

**Reply by azrikahar** [reply]
Everything seems fine, but two things of note:

- Step 5 should be `npm run cli bootstrap --workspace=api` for SQLite specifically, as described in the docs: https://docs.directus.io/contributing/running-locally/#_6-initialize-the-database. This means your `data.db` file should live in `/api`, and not the project root.

- Step 6 only runs the App (Vue), but you will also need to run the Api (Express) as well, which is `npm run dev -w directus` 👍 Make sure to run _both_ of them (separate terminals) as described in my previous reply above. Link to docs: https://docs.directus.io/contributing/running-locally/#_7-start-the-development-server
